pkg/roachprod/roachprod.go

```diff
@@ -90,17 +102,29 @@ func verifyClusterName(l *logger.Logger, clusterName, username string) error {
 	// Use the vm.Provider account names, or --username.
 	var accounts []string
 	if len(username) > 0 {
-		accounts = []string{username}
+		cleanAccount := vm.DNSSafeName(username)
+		if cleanAccount != username {
+			l.Printf("WARN: using `%s' as username instead of `%s'", cleanAccount, username)
+		}
+		accounts = []string{cleanAccount}
 	} else {
 		seenAccounts := map[string]bool{}
-		active, err := vm.FindActiveAccounts(l)
+		var active map[string]string
+		var err error
+		if _findActiveAccounts != nil {
```

Comment:
Can elide this check and just use `active, err = _findActiveAccounts(l)` as per above comment.